src/libs/actions/Policy.js

```diff
@@ -741,17 +746,14 @@ function generateDefaultWorkspaceName(email = '') {
         return defaultWorkspaceName;
     }
 
-    // Check if this name already exists in the policies
-    let suffix = 0;
-    _.forEach(allPolicies, (policy) => {
-        const name = lodashGet(policy, 'name', '');
-
-        if (name.toLowerCase().includes(defaultWorkspaceName.toLowerCase())) {
-            suffix += 1;
-        }
-    });
-
-    return suffix > 0 ? `${defaultWorkspaceName} ${suffix}` : defaultWorkspaceName;
+    // find default named workspaces and increment the last number
+    const numberRegEx = new RegExp(`${defaultWorkspaceName} ?(\\d*)`, 'i');
+    const defaultWorkspaceNumbers = _.chain(allPolicies)
+        .filter(policy => policy.name && numberRegEx.test(policy.name))
+        .map(policy => parseInt(numberRegEx.exec(policy.name)[1] || 1, 10)) // parse the number at the end
+        .value();
+    const lastNumber = defaultWorkspaceNumbers.sort().pop();
```

Comment:
I was thinking we might be able to simplify/avoid this line by using `.max()` on the chain above, but I don't know if it would work when you don't have any workspaces, so maybe it's best to keep like you have it now.

Reply:
I like this I included a change for it. `max()` returns `-Infinity` when the list is empty so I just checked for that.